src/main/scala/edu/scalanus/interpreter/ScalanusInterpreter.scala

```diff
@@ -0,0 +1,187 @@
+    case irExpr: IrExpr => evalExpr(irExpr, context, scope)
+  }
+
+  def evalAssignStmt(irAssignStmt: IrAssignStmt, context: ScalanusScriptContext, scope: Int): Any = ???
+
+
+  //
+  // Patterns
+  //
+
+  def evalPattern(irPattern: IrPattern, context: ScalanusScriptContext, scope: Int): Any = ???
+
+  def evalSimplePattern(irSimplePattern: IrSimplePattern, context: ScalanusScriptContext, scope: Int): Any =
+    irSimplePattern match {
+      case irWildcardPattern: IrWildcardPattern => evalWildcardPattern(irWildcardPattern, context, scope)
+      case irRefPattern: IrRefPattern => evalRefPattern(irRefPattern, context, scope)
+      case irValuePattern: IrValuePattern => evalValuePattern(irValuePattern, context, scope)
+    }
+
+  def evalWildcardPattern(irWildcardPattern: IrWildcardPattern, context: ScalanusScriptContext, scope: Int): Any = ???
+
+  def evalRefPattern(irRefPattern: IrRefPattern, context: ScalanusScriptContext, scope: Int): Any = ???
+
+  def evalValuePattern(irValuePattern: IrValuePattern, context: ScalanusScriptContext, scope: Int): Any = ???
+
+
+  //
+  // Items
+  //
+
+  def evalItem(irItem: IrItem, context: ScalanusScriptContext, scope: Int): Any = irItem match{
+    case irFnItem: IrFnItem => evalFnItem(irFnItem, context, scope)
+  }
+
+  def evalFnItem(irFnItem: IrFnItem, context: ScalanusScriptContext, scope: Int): Any = ???
+
+
+  //
+  // Expressions
+  //
+
+  def evalExpr(irExpr: IrExpr, context: ScalanusScriptContext, scope: Int): Any = {
+    irExpr match{
+      case irBlock: IrBlock => evalBlock(irBlock, context, scope)
+      case irRefExpr: IrRefExpr => evalRefExpr(irRefExpr, context, scope)
+      case irValue: IrValue => evalValue(irValue, context, scope)
+      case irUnaryExpr: IrUnaryExpr => evalUnaryExpr(irUnaryExpr, context, scope)
+      case irIncrExpr: IrIncrExpr => evalIncrExpr(irIncrExpr, context, scope)
+      case irBinaryExpr: IrBinaryExpr => evalBinaryExpr(irBinaryExpr, context, scope)
+      case irFnCallExpr: IrFnCallExpr => evalFnCallExpr(irFnCallExpr, context, scope)
+      case irForExpr: IrForExpr => evalForExpr(irForExpr, context, scope)
+      case irWhileExpr: IrWhileExpr => evalWhileExpr(irWhileExpr, context, scope)
+      case irLoopExpr: IrLoopExpr => evalLoopExpr(irLoopExpr, context, scope)
+      case irIfExpr: IrIfExpr => evalIfExpr(irIfExpr, context, scope)
+      case irBreak: IrBreak => evalBreak(irBreak, context, scope)
+      case irContinue: IrContinue => eval(irContinue, context, scope)
+      case irReturn: IrReturn => eval(irReturn, context, scope)
+      case irTuple: IrTuple => evalTuple(irTuple, context, scope)
+      case irDict: IrDict => evalDict(irDict, context, scope)
+    }
+  }
+
+  def evalBlock(irBlock: IrBlock, context: ScalanusScriptContext, scope: Int): Any =
+    irBlock.stmts.foldLeft[Any] (Unit) { (_, stmt) => evalStmt(stmt, context, scope) }
+
+  def evalRefExpr(irRefExpr: IrRefExpr, context: ScalanusScriptContext, scope: Int): Any = ???
```

Comment:
jeśli tak to nie wiem po co IrRef nie jest roszerzeniem IrExpr skoro nigdy się nie pojawa samo :P